### PR TITLE
Support SSL connection

### PR DESCRIPTION
**Pull request by jbpoulsen**
An optional parameter when creating a new database connection enables users to enter the path of a SSL certificate file.

image

### CODE REVIEW

File: src/mysqlTreeDataProvider.ts [hunk 1]
```diff
@@ -49,6 +49,8 @@ export class MySQLTreeDataProvider implements vscode.TreeDataProvider<INode> {
             return;
         }
 
+        const certPath = await vscode.window.showInputBox({ prompt: "SSL certificate path. Empty to ignore", placeHolder: "certificate file path", ignoreFocusOut: true, value: "",  });
```
Comment (formulahendry) [inline src/mysqlTreeDataProvider.ts]: if (certPath === undefined) {
            return;
        }

File: src/model/tableNode.ts [hunk 2]
```diff
@@ -9,7 +9,8 @@ import { INode } from "./INode";
 
 export class TableNode implements INode {
     constructor(private readonly host: string, private readonly user: string, private readonly password: string,
```
Comment (formulahendry) [inline src/model/tableNode.ts]: Also update mysql.createConnection in this file

File: src/common/utility.ts [hunk 3]
```diff
@@ -93,4 +94,29 @@ export class Utility {
             setTimeout(resolve, ms);
         });
     }
+
+    public static createConnection(connectionOptions: IConnection) : any
+    {
+        var connection;
+        if(connectionOptions.certPath != '' && fs.existsSync(connectionOptions.certPath)){
```
Comment (formulahendry) [inline src/common/utility.ts]: !connectionOptions.certPath

File: src/mysqlTreeDataProvider.ts (hunk 1 above)
Comment (formulahendry) [reply src/mysqlTreeDataProvider.ts]: No need to add `value: ""`

Comment (formulahendry) [reply src/mysqlTreeDataProvider.ts]: `prompt: "[Optional] SSL certificate path. Leave empty to ignore"`

File: src/model/tableNode.ts (hunk 2 above)
Comment (jbpoulsen) [reply src/model/tableNode.ts]: Not sure what you mean here, there is no mysql.createConnection call in this file?

Comment (formulahendry) [reply src/model/tableNode.ts]: It's here: https://github.com/formulahendry/vscode-mysql/blob/5d28e182fb9be7bb0c5fd49e1b96a172760e0c98/src/model/tableNode.ts#L27

Comment (jbpoulsen) [reply src/model/tableNode.ts]: Oh well, that was changed from another pull request just recently.. ill make the changes

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,6 @@
+## 0.1.1 (2017-12-03)
```
Comment (formulahendry) [inline CHANGELOG.md]: Not sure why GitHub show previous changes...

File: src/common/utility.ts [hunk 5]
```diff
@@ -79,6 +80,29 @@ export class Utility {
         return vscode.window.showTextDocument(textDocument);
     }
 
+    public static createConnection(connectionOptions: IConnection): any {
+        let connection;
+        if (!connectionOptions.certPath && fs.existsSync(connectionOptions.certPath)) {
```
Comment (formulahendry) [inline src/common/utility.ts]: Sorry, should be `connectionOptions.certPath`

File: CHANGELOG.md (hunk 4 above)
Comment (jbpoulsen) [reply CHANGELOG.md]: I'll see if i can figure it out..
Learning GIT is not the easiest thing in the world, thats for sure.

**Comment by jbpoulsen** [comment]
Alright, i finally think i figured it all out.
Sorry for all the confusion here on Github.. i'm still learning this.
Let me know if you need anything else from me for this feature to work.

**Pull request merged**